components/selectors/parser.rs

```diff
@@ -120,7 +120,7 @@ pub trait Parser {
     }
 }
 
-#[derive(PartialEq, Clone, Debug)]
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
```

Comment:
Is this change needed just due to the bounds in `SelectorImpl`s definition?

If so, I believe we can remove them since we switched to conditional compilation in `style` and should have a concrete type now.

Also, in pseudo-classes it doesn't really matter I think, since we only store `PseudoElement`s in `HashMap`s.